oidc-service-impl/src/test/java/io/mosip/esignet/services/ConsentHelperServiceTest.java

```diff
@@ -258,6 +371,112 @@ public void processConsent_withEmptyConsent_thenPass(){
+
+        List<String> acceptedClaims = consentDetail.getAcceptedClaims();
+        List<String> permittedScopes = consentDetail.getPermittedScopes();
+        Collections.sort(acceptedClaims);
+        Collections.sort(permittedScopes);
+        Map<String,Object> payLoadMap = new TreeMap<>();
+        payLoadMap.put("accepted_claims",acceptedClaims);
+        payLoadMap.put("permitted_authorized_scopes",permittedScopes);
+
+        String signature = generateSignature(payLoadMap);
+        consentDetail.setSignature(signature);
+        consentDetail.setPsuToken("psutoken");
+
+        PublicKeyRegistry publicKeyRegistry =new PublicKeyRegistry();
+        publicKeyRegistry.setCertificate(certificateString);
+        Mockito.when(authorizationHelperService.getIndividualId(oidcTransaction)).thenReturn("individualId");
+        Mockito.when(publicKeyRegistryService.findFirstByIdHashAndThumbprintAndExpiredtimes(Mockito.any(),Mockito.any())).thenReturn(Optional.empty());
+
+        Mockito.when(consentService.getUserConsent(userConsentRequest)).thenReturn(Optional.of(consentDetail));
+
+        consentHelperService.processConsent(oidcTransaction,true);
+
+        Assert.assertEquals(oidcTransaction.getConsentAction(),ConsentAction.CAPTURE);
+    }
+
+    @Test
+    public void processConsent_withInvalidSignature_thenFail(){
+
+        OIDCTransaction oidcTransaction=new OIDCTransaction();
+        oidcTransaction.setClientId("abc");
+        oidcTransaction.setPartnerSpecificUserToken("123");
+        oidcTransaction.setVoluntaryClaims(List.of("email"));
+        oidcTransaction.setEssentialClaims(List.of());
+        oidcTransaction.setRequestedAuthorizeScopes(List.of());
+        UserConsentRequest userConsentRequest = new UserConsentRequest();
+        userConsentRequest.setClientId(oidcTransaction.getClientId());
+        userConsentRequest.setPsuToken(oidcTransaction.getPartnerSpecificUserToken());
+
+        ConsentDetail consentDetail=new ConsentDetail();
+        consentDetail.setAcceptedClaims(Arrays.asList());
+        consentDetail.setSignature("haa.naa");
+
+        Mockito.when(consentService.getUserConsent(userConsentRequest)).thenReturn(Optional.of(consentDetail));
+
+        try{
+            consentHelperService.processConsent(oidcTransaction,true);
+            Assert.fail();
+        }catch (Exception e){
+            Assert.assertEquals(e.getMessage(),ErrorConstants.INVALID_AUTH_TOKEN);
+        }
+    }
+
+    @Test
+    public void toGenerateTestingSignatrue() throws Exception {
+       // Payload payload = new Payload();
+        List<String> acceptedClaims =Arrays.asList("name","email","gender");
+        List<String> permittedScopes =new ArrayList<>();
+        Collections.sort(acceptedClaims);
+        Collections.sort(permittedScopes);
+        Map<String,Object> payLoadMap = new TreeMap<>();
+        payLoadMap.put("permitted_authorized_scopes",permittedScopes);
+        payLoadMap.put("accepted_claims",acceptedClaims);
```

Comment:
this test case looks like half way, which method are we testing ?

Reply:
not important so removed